pontoon/sync/tasks.py

```diff
@@ -128,13 +128,10 @@ def sync_project(
         full_scan=force
     )
 
-    new_entities = new_entities.result
+    new_entities = source_changes.get('new_entities')
 
-    if source_changes.get('new_entities'):
-        new_entities = new_entities + source_changes.get('new_entities')
-
-    if db_project.pretranslation_enabled and len(new_entities):
-        new_entities = list(set(new_entities))  # take only unique entities
+    if db_project.pretranslation_enabled and new_entities:
+        new_entities = list(set(source_changes.get('new_entities')))
```

Comment:
Can't you move this line above the `if` and remove line 131?

Reply:
For uniformity, we return None and the set() doesn't take non-iterables. So first we need to check the case of value not being none and then proceed.